fipy/matrices/sparseMatrix.py

```diff
@@ -119,8 +119,176 @@ def addAtDiagonal(self, vector):
     def exportMmf(self, filename):
         pass
 
+    @property
+    def CSR(self):
+        pass
+
+    @property
+    def LIL(self):
+        pass
+
+    @property
+    def T(self):
+        pass
+
 ##     def __array__(self):
 ##      shape = self._shape
 ##      indices = numerix.indices(shape)
 ##         numMatrix = self.take(indices[0].ravel(), indices[1].ravel())
 ##      return numerix.reshape(numMatrix, shape)
+
+class _Mesh2Matrix(object):
+    def __init__(self, mesh, numberOfEquations=1, numberOfVariables=1,
+                 orderer=lambda IDs: IDs):
+        """Creates a mapping between mesh cells and matrix rows and columns
+
+        Parameters
+        ----------
+        mesh : ~fipy.meshes.mesh.Mesh
+            The `Mesh` to be mapped to a matrix
+        numberOfVariables : int
+            The local columns of the matrix are determined by
+            `numberOfVariables * len(mesh._localNonOverlappingCellIDs)`.
+        numberOfEquations : int
+            The local rows of the matrix are determined by
+            `numberOfEquations * len(mesh._localNonOverlappingCellIDs)`.
+        orderer : fn
+            Function that changes the order of IDs to satisfy matrix
+            construction requirements (Default returns IDs unchanged).
+        """
+        self.mesh = mesh
+        self.numberOfVariables = numberOfVariables
+        self.numberOfEquations = numberOfEquations
+        self.orderer = orderer
+
+    def _cellIDsToGlobalRowIDs(self, IDs):
+        N = len(IDs)
+        M = self.numberOfEquations
+        return (numerix.vstack([IDs] * M) + numerix.indices((M,N))[0] * self.mesh.globalNumberOfCells).flatten()
```

Comment:
There are four instances of the same expression (vstack + indices + cells). Might be worth making one expression with different arguments.

Reply:
Refactored